bokeh/models/widgets/sliders.py

```diff
@@ -110,6 +116,13 @@ def _check_missing_dimension(self):
             if self.start == self.end:
                 return '{!s} with title {!s}'.format(self, self.title)
 
+
+    @warning(SLIDER_CALLBACK_APPLY_TO_CUSTOM_JS)
+    def _check_callback_policy_called_with_js_custom(self):
+        if hasattr(self, 'callback_policy') and not hasattr(self, 'callback'):
+            return 'Callback policy currently apply to JS callbacks'
```

Comment:
Hi @slamer59 as I mentioned on gitter, just checking for `callback` is too restrictive. There is also now `js_on_change('value', ...)` which is actually what I would regard as current best practice. Those are stored here in the `. js_property_callbacks` property on all models:

https://github.com/bokeh/bokeh/blob/master/bokeh/model.py#L335

That would need to be checked for a key `'value'`.

However, I also in looking at this that that is not right either. Because it should be harmless to set a callback policy with *no callbacks* but that is not the case. What really would need to happen is that the Model needs to be checked for the positive presence of a Python callback on `"value"` That's unfortunately a bit down in the weeds to accomplish. I am happy to try and provide more guidance if you want to pursue it, but also happy for a PR that just updates the docs and nothing else.

Reply:
And thinking about it even more, I am not sure it is appropriate to warn at all. There can well be valid use cases for having a callback policy to affect a JS callback and at the same time having a python callback that will be known and expected to run every change. I think for now just a documentation update is the best course.

Reply:
Oh. I totally miss your point on this. My intention was more to write something to check if callback is a CustomJS instance:

```
if hasattr(self, 'callback_policy') and not isinstance(callback, CustomJS):
```

I'm not sure to understand all machinery and will just update documentation consequently.

Thanks